src/js/api/auth/login.test.js

```diff
@@ -0,0 +1,64 @@
+import { login } from "./login.js"
+import { apiPath } from "../constants.js"
+import { save } from "../../storage/index.js"
+
+global.fetch = jest.fn()
+
+jest.mock("../headers.js", () => ({
+  headers: jest.fn((contentType) => {
+    if (contentType === "application/json") {
+      return { "Content-Type": "application/json" }
+    }
+    throw new Error("Invalid content type")
+  }),
+}))
+
+jest.mock("../../storage/index.js", () => ({
+  save: jest.fn(),
+}))
+
+describe("login function", () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it("logs in a user and saves profile when provided data is valid", async () => {
+    fetch.mockResolvedValueOnce({
+      ok: true,
+      json: async () => ({
+        accessToken: "mockedAccessToken",
+        user: { id: "666", name: "Example User" },
+      }),
+    })
+
+    const email = "test@example.com"
+    const password = "password123"
+
+    const profile = await login(email, password)
+
+    expect(fetch).toHaveBeenCalledWith(`${apiPath}/social/auth/login`, {
+      method: "post",
+      body: JSON.stringify({ email, password }),
+      headers: { "Content-Type": "application/json" },
+    })
+
+    expect(save).toHaveBeenCalledWith("token", "mockedAccessToken")
+    expect(save).toHaveBeenCalledWith("profile", {
+      user: { id: "666", name: "Example User" },
+    })
+
+    expect(profile).toEqual({ user: { id: "666", name: "Example User" } })
+  })
+
```

Comment:
Hi Weronika! Your code looks great. I was just wondering if there should be test that checks if the token gets  deleted. I see that the login function apparently deletes the token at some point. This is from the login function: "if (response.ok) {
		const profile = await response.json();
		save("token", profile.accessToken);
		delete profile.accessToken;
		save("profile", profile);
		return profile;
	}". This is only a suggestion, I am not sure myself if this is needed. Great work, Weronika.   Kind regards,

Alex

Reply:
In reality, you have "overtested" here - meaning that you went above and beyond what we asked. It isn't bad, just saying that you can compelte this test with very little code.